src/Language/Haskell/Liquid/Types/Types.hs

```diff
@@ -1522,8 +1560,10 @@ mapExprReft f = mapReft g
   where
     g (MkUReft (F.Reft (x, e)) p) = MkUReft (F.Reft (x, f x e)) p
 
+-- const False (not dropping dict) is probably fine since there will not be refinement on
+-- dictionaries
 isTrivial :: (F.Reftable r, TyConable c) => RType c tv r -> Bool
-isTrivial t = foldReft False (\_ r b -> F.isTauto r && b) True t
+isTrivial t = foldReft (const False) False (\_ r b -> F.isTauto r && b) True t
```

Comment:
`isTrivial` should be flagged. `const False` means foldReft will not erase anything. Changing `const False` to `isClass` might fix some tests